AdminUi/apps/admin_ui/lib/home/clients_overview/modals/remove_clients_dialog.dart

```diff
@@ -0,0 +1,132 @@
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+import 'package:go_router/go_router.dart';
+import 'package:logger/logger.dart';
+
+import '/core/core.dart';
+
+Future<void> showRemoveClientsDialog({
+  required BuildContext context,
+  required Set<String> selectedClients,
+  required VoidCallback onClientsRemoved,
+}) async {
+  await showDialog<void>(
+    context: context,
+    builder: (BuildContext context) => _RemoveClientsDialog(selectedClients: selectedClients, onClientsRemoved: onClientsRemoved),
+  );
+}
+
+class _RemoveClientsDialog extends StatefulWidget {
+  final Set<String> selectedClients;
+  final VoidCallback onClientsRemoved;
+
+  const _RemoveClientsDialog({required this.selectedClients, required this.onClientsRemoved});
+
+  @override
+  State<_RemoveClientsDialog> createState() => _RemoveClientsDialogState();
+}
+
+class _RemoveClientsDialogState extends State<_RemoveClientsDialog> {
+  bool _deleting = false;
+  bool _deletionSucceeded = false;
+  String? _deletionError;
+  final _deletedClients = <String>[];
+
+  @override
+  Widget build(BuildContext context) {
+    return PopScope(
+      canPop: !_deleting,
+      child: AlertDialog(
+        title: Text(widget.selectedClients.length == 1 ? 'Delete Client' : 'Delete Clients', textAlign: TextAlign.center),
+        content: switch ((_deleting, _deletionSucceeded)) {
+          (_, true) => _RemoveClientsSucceeded(numberOfDeletedClients: _deletedClients.length),
+          (true, _) => const _RemoveClientsLoading(),
+          (_, _) => Text(
+              _deletionError ?? 'Are you sure you want to delete the selected ${widget.selectedClients.length > 1 ? 'clients' : 'client'}?',
+              style: TextStyle(color: Theme.of(context).colorScheme.primary),
+            ),
+        },
+        actions: <Widget>[
+          OutlinedButton(
+            onPressed: _deleting ? null : () => context.pop(),
+            child: Text(_deletionSucceeded ? 'Close' : 'Cancel'),
+          ),
+          if (!_deletionSucceeded)
+            TextButton(
+              onPressed: _deleting ? null : _deleteSelectedClients,
+              child: Text('Yes', style: TextStyle(color: Theme.of(context).colorScheme.error)),
+            ),
+        ],
+      ),
+    );
+  }
+
+  Future<void> _deleteSelectedClients() async {
+    setState(() => _deleting = true);
+
+    try {
+      for (final clientId in widget.selectedClients.toList()) {
+        await GetIt.I.get<AdminApiClient>().clients.deleteClient(clientId);
+        widget.onClientsRemoved();
+        widget.selectedClients.remove(clientId);
+        _deletedClients.add(clientId);
+      }
+    } catch (e) {
+      GetIt.I.get<Logger>().e('An error occurred while deleting the client(s). $e');
+
+      setState(() {
+        _deleting = false;
+        _deletionSucceeded = false;
+        _deletionError = 'An error occurred while deleting the client(s). Please try again.';
```

Comment:
hardcoded text in setState is IMO not good. Same as in `change_client_secret_dialog.dart`

Reply:
As an alternative we would use a translated text here. But we don't want to have translations so this is not used here.